code/lib/types/src/modules/csf.ts

```diff
@@ -135,13 +138,15 @@ export type ViewMode = ViewModeBase | 'story' | 'info' | 'settings' | string | u
 
 type Layout = 'centered' | 'fullscreen' | 'padded' | 'none';
 
-export interface Parameters extends ParametersBase {
-  fileName?: string;
+export interface StorybookParameters {
   options?: Addon_OptionsParameter;
   /** The layout property defines basic styles added to the preview body where the story is rendered. If you pass 'none', no styles are applied. */
   layout?: Layout;
-  docsOnly?: boolean;
-  [key: string]: any;
+}
+
+export interface StorybookInternalParameters extends StorybookParameters {
+  fileName?: string;
+  docsOnly?: true;
 }
```

Comment:
So this seems a bit tricky. We don't want users to *set* `fileName` or `docsOnly`, but they might want to *read* them, eg. in decorators? How should we recommend that gets handled in types?

Reply:
My idea was that even reading them from the `StoryContext` might be more something that addons do in decorators, but not users do in decorators. Because in that case, this is something that only addon authors could import.

Otherwise, this might become really tricky indeed. We could make 2 different types. `ParametersInput` for stuff that users can set, and `Parameters` for what can be read in the context. But I'm not sure if it is worth the effort. 

There is something to say for that approach as well when it comes to making stuff optional in `ParametersInput`, and non-nullable in `Parameters`. Because there is a default value set in the addon. 

Another approach to "solve this" might be to just use `Partial<Parameters>` for `ParametersInput` as I believe that parameters should always be optional in the "input of a story". This kind of leaves this task to the addon author to make sure that parameters always get a default value.

Add some point, we could check if this is really true, once we import the preview file of an addon in the `preview.ts`:

```ts
// preview.ts
import * as DocsAddon from '@storybook/addon-docs/preview';
import * as BackgroundsAddon from '@storybook/addon-backgrounds/preview';

export default {
  addons: [DocsAddon, BackgroundsAddon],
} satisfies PreviewConfig;
```

Reply:
I leave it like this for now, let's revisit this once we are gonna improve the types of Parameters